storybookjs/storybook
# Angular build time optimization

Issue: https://github.com/storybooks/storybook/issues/3968

## What I did

**Note:** This is work-in-progress

- Renamed `loadTsConfig`  to `getTsLoaderOptions` because that is what it actually does (https://github.com/storybooks/storybook/commit/58010575192ec05cc061f7084f9cbb62240addb8#diff-aaa47dcc2ee48513f45210bb35a92d4cR1)
- Set `transpileOnly: true` as default option for `ts-loader`
- Added `fork-ts-checker-webpack-plugin` as a dependency in `app/angular`
- Use `ForkTsChecker` and set option `tsconfig` to the same that is used for `ts-loader` (using `getTsLoaderOptions`)

## Checklist

- [x] Set `transpileOnly: true` as default `ts-loader` option
- [x] Add `ForkTsChecker` to the list of plugins
- [x] Verify it works
- [x] Check, why `ForkTsChecker` does check `jest` (`ERROR in D:/04_Development/ng-projects/storybook/node_modules/@types/jest/index.d.ts(1103,9):` and many more in this file)
- [x] Update test-case `ts_config.test.js`

## How to test

Is this testable with Jest or Chromatic screenshots? **yes**
Does this need a new example in the kitchen sink apps? **no**
Does this need an update to the documentation? **yes**

If your answer is yes to any of these, please make sure to include it in your PR.

**Manual test:** 
Start `yarn storybook` in `examples/angular-cli/`

Using `ForkTsChecker` adds the folloing message when starting storybook
```
Starting type checking service...
Using 1 worker with 2048MB memory limit
```


For maintainers only: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
